cmd/client-s3.go

```diff
@@ -384,6 +386,61 @@ func (c *s3Client) ListNotificationConfigs(arn string) ([]notificationConfig, *p
 	return configs, nil
 }
 
+//
+func (c *s3Client) Select(expression, sseKey string) (io.ReadCloser, *probe.Error) {
+	bucket, object := c.url2BucketAndObject()
+	origContentType := mime.TypeByExtension(filepath.Ext(strings.TrimSuffix(strings.TrimSuffix(object, ".gz"), ".bz2")))
+	contentType := mime.TypeByExtension(filepath.Ext(object))
+	opts := minio.SelectObjectOptions{
+		Expression:     expression,
+		ExpressionType: minio.QueryExpressionTypeSQL,
+	}
+	if strings.Contains(origContentType, "csv") {
+		opts.InputSerialization = minio.SelectObjectInputSerialization{
+			CompressionType: minio.SelectCompressionNONE,
+			CSV: &minio.CSVInputOptions{
+				FileHeaderInfo:  minio.CSVFileHeaderInfoUse,
+				RecordDelimiter: "\n",
+				FieldDelimiter:  ",",
+			},
+		}
+		opts.OutputSerialization = minio.SelectObjectOutputSerialization{
+			CSV: &minio.CSVOutputOptions{
+				RecordDelimiter: "\n",
+				FieldDelimiter:  ",",
+			},
+		}
+	} else if strings.Contains(origContentType, "json") {
```

Comment:
we can maybe skip if the file is not either one of `csv` and `json`

Reply:
it is skipped not by this function instead by the caller...

Reply:
This is the output I see when I run the recursive command with one file which is neither csv not json
```
mc: <ERROR> Unable to run select Invalid data source type. Only CSV and JSON are supported at this time.
2.000000
1178008.000000
```
I think this tells the user that they have a non `csv and json` file. I guess this is good enough.